### ISSUE TITLE
Improve behavior of sortedResultsUsingDescriptors when passing an empty array

### ISSUE THREAD

**Issue by jpsim** [realm/realm-cocoa#4119]
I think sorting should be a no-op when no descriptors are passed. We could alternatively throw an exception with a more useful error message, but I think sorting with no descriptors should be the same as sorting with the identity (e.g. not sorting at all).

```
views.cpp:36: [realm-core-1.5.0] Assertion failed: !column_indices.empty()
0   Realm                               0x00000001129beaa8 _ZN5realm4utilL18terminate_internalERNSt3__118basic_stringstreamIcNS1_11char_traitsIcEENS1_9allocatorIcEEEE + 40
1   Realm                               0x00000001129bedaa _ZN5realm4util9terminateEPKcS2_lOSt16initializer_listINS0_9PrintableEE + 506
2   Realm                               0x0000000112b6950c _ZN5realm14SortDescriptorC2ERKNS_5TableENSt3__16vectorINS5_ImNS4_9allocatorImEEEENS6_IS8_EEEENS5_IbNS6_IbEEEE + 1244
3   Realm                               0x0000000112931754 _Z32RLMSortDescriptorFromDescriptorsRN5realm5TableEP7NSArrayIP17RLMSortDescriptorE + 1422
4   Realm                               0x0000000112997506 -[RLMResults sortedResultsUsingDescriptors:] + 101
5   RealmSwift                          0x0000000112dd748b _TFC10RealmSwift7Results6sorteduRd__s8SequenceWd__8Iterator7Element_zVS_14SortDescriptorrfT2byqd___GS0_x_ + 251
```

**Comment by bdash** [comment]
@tgoyne submitted a fix for this in #4118.

**Comment by jpsim** [comment]
Great!
